sunpy/io/tests/test_cdf.py

```diff
@@ -23,3 +23,11 @@ def test_read_cdf():
     assert col.unit == u.Unit("1 / (cm2 MeV s sr)")
     # Check that fillvals are replaced by NaN
     assert np.sum(np.isnan(col)) == 189
+
+
+def test_check_nan_values():
```

Comment:
Did this test fail before the code change?

Reply:
No before it was working fine , now after the latest commit it is failing

Reply:
![Screenshot 2024-12-13 000016](https://github.com/user-attachments/assets/eeefb742-7829-42a5-8f48-a351cc379a53)

Reply:
That needs to be fixed but that was not what I was referring to. 

The goal behind the unit test is to make sure that we test the original bug is fixed.

If this test passes before and after the bugfix, this test is useless. 

We need a unit test that fails before the bugfix change and then will pass after said change.

Can you write one?